File: GameLibrary/src/cc/lib/zombicide/ZAttackType.java
package cc.lib.zombicide;

import cc.lib.annotation.Keep;

@Keep
public enum ZAttackType {
    NORMAL,
    FIRE,
    ELECTROCUTION,
    DISINTEGRATION,
    BLADE,
    CRUSH,
    RANGED_ARROWS,
    RANGED_BOLTS,
    RANGED_THROW,
    EARTHQUAKE,
    MENTAL_STRIKE,
    DRAGON_FIRE;

    public boolean needsReload() {
        return this == RANGED_BOLTS;
    }
}

File: Zombicide/src/cc/lib/zombicide/ZAttackType.kt
package cc.lib.zombicide

import cc.lib.annotation.Keep

@Keep
enum class ZAttackType(val description: String) {
	FAMILIAR("Mauled"),
	FIRE("Burned"),
	ELECTROCUTION("Electrocuted"),
	DISINTEGRATION("Disintegrated"),
	BLADE("Impaled"),
	CRUSH("Crushed"),
	RANGED_ARROWS("Sniped"),
	RANGED_BOLTS("Sniped"),
	RANGED_THROW(""),
	EARTHQUAKE("Earthquake"),
	MENTAL_STRIKE("Scanned"),
	DRAGON_FIRE("Incinerated"),
	FREEZE("Frozen"),
	FRIENDLY_FIRE("Friendly Fire"),
}